src/actions.js

```diff
@@ -36,11 +36,13 @@ module.exports = {
 						id: 'gpi',
 						default: '1',
 						regex: Regex.NUMBER,
+						useVariables: true,
 					},
 				],
 				callback: async (event) => {
 					let opt = event.options
-					var gpi = parseInt(opt.gpi)
+					let gpiText = await parseVariable(opt.gpi)
```

Comment:
And this to use it, which gives some additional benefits (I think local variables and stuff):
```suggestion
					let gpiText = await context.parseVariable(opt.gpi)
```

Reply:
Can you point me to documentation or any reference to this context.parsevariable function? I tested it and companion does not recognize it as anything, and I cannot find any documentation regarding it. Can you also please detail what specific functionality it provides over the method I already implemented?

Reply:
I'm struggling to find some docs about why, but you can see it here:
https://bitfocus.github.io/companion-module-base/interfaces/CompanionActionDefinition.html#callback
https://bitfocus.github.io/companion-module-base/types/CompanionActionContext.html

And the code:
https://github.com/bitfocus/companion-module-base/blob/main/src/module-api/common.ts

But from what I understood it was good practise. You may need to double-check on Slack...

Reply:
Unless I can find a good reason for doing it, this is outside of the scope of my changes. I only am looking to allow for custom variables to be used, and am not the maintainer of this module. Maybe someone with more experience can add this at a later date.

Reply:
> this is outside of the scope of my changes

But this is part of adding variable parsing!

@Julusian or others will know if this is required or good practise or what (and may have a link to some better docs or be able to update the wiki to advise).

Reply:
It looks like the `@companion-module/base` library in this module would need updating to be able to use the context version of variable parsing.

We don't require that context version to be used, but doing it allows variables like `$(this:row)` to be used too, so it is good practise. 
Companion 4.1 will likely expand this to more variables, but also might negate the need for doing this entirely

Reply:
It was done to "@companion-module/base": "^1.5.0" in this PR, is that not recent enough?

It seems to have the context in that version:
https://github.com/bitfocus/companion-module-base/blob/v1.5.0/src/module-api/action.ts#L40

Reply:
oh I missed that. and yarn picked `"1.11.3"`, so it definitely will be available

Reply:
I couldn't get companion to recognize it, so I will probably go without for now. Someone else may be able to make that work. Going to leave these changes as made.

Reply:
That's odd, had you added the other line to get the context too that you'd previously resolved from my other comment? https://github.com/bitfocus/companion-module-rossvideo-rosstalk/pull/40#discussion_r2052731129

Reply:
Yes I went through each of your comments one by one and implemented them. I will reattempt it just to make sure I didn't miss something.

Reply:
Okay I reattempted it from scratch, without looking at anything I did the last time for reference. I got the exact same issue, Instance/Wrapper/rosstalk: Error executing action: context.parseVariable is not a function. Any ideas?

Reply:
That's odd @jreynolds67 . Can you push your broken attempt to a new branch or something so we can try and have a look at it please?

Reply:
Done @peternewman - https://github.com/jreynolds67/rosstalk-test

Reply:
Thanks, if you were able to do it as another branch rather than a separate repo it makes seeing what's changed easier, but I'll try and look anyway.

Which one does it fail on, as you've got some like `gpi` calling it directly, but you've also got your own parseVariable function (which appears to be unused), although I don't think that should be confusing it:
https://github.com/jreynolds67/rosstalk-test/blob/7dfa251f4b59353f576561a0b3617b487085eb65/src/actions.js#L22-L27

Do you get anything more verbose in the logs or console?

Reply:
Oops sorry, not super familiar with GitHub, my bad. My latest test I did it from the custom command. However, when I couldn't get it to work the first time, I went down the list of every type action available and they all gave the same error.

Reply:
Apologies, you need to use the function `parseVariablesInString` ! Sorry I hadn't clocked before there was a local function which used to be calling self.parseVariablesInString, hence you used a different function name.

So drop your local function to avoid any more confusion (as it shouldn't be necessary and I'm sure they'll make that appropriate optimisation centrally) and then swap each `context.parseVariable` to `context.parseVariablesInString` and you should be good finally. Sorry for not spotting it earlier!

Reply:
That was the problem, its working great now.

Reply:
Great, glad to hear it and sorry for not spotting it earlier.